ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -22,32 +22,56 @@ use tabby_common::{
 use tokio_cron_scheduler::{Job, JobScheduler};
 use tower::ServiceExt;
 use tower_http::services::ServeDir;
-use tracing::debug;
+use tracing::{debug, error};
+
+use crate::schema::repository::RepositoryService;
 
-#[derive(Debug)]
 pub struct RepositoryCache {
-    repositories: RwLock<HashMap<RepositoryKey, RepositoryMeta>>,
-    configured_repositories: Vec<RepositoryConfig>,
+    repository_lookup: RwLock<HashMap<RepositoryKey, RepositoryMeta>>,
+    repositories: RwLock<Vec<RepositoryConfig>>,
+    service: Arc<dyn RepositoryService>,
+}
+
+impl std::fmt::Debug for RepositoryCache {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("RepositoryCache")
+            .field("repository_lookup", &self.repository_lookup)
+            .field("repositories", &self.repositories)
+            .finish()
+    }
 }
 
 impl RepositoryCache {
-    pub fn new_initialized(configured_repositories: Vec<RepositoryConfig>) -> RepositoryCache {
+    pub async fn new_initialized(service: Arc<dyn RepositoryService>) -> RepositoryCache {
         let cache = RepositoryCache {
+            repository_lookup: Default::default(),
             repositories: Default::default(),
-            configured_repositories,
+            service,
+        };
+        if let Err(e) = cache.reload().await {
+            error!("Failed to load repositories: {e}");
         };
-        cache.reload();
         cache
     }
 
-    fn reload(&self) {
-        let mut repositories = self.repositories.write().unwrap();
+    async fn reload(&self) -> Result<()> {
+        let new_repositories = self
+            .service
+            .list_repositories(None, None, None, None)
+            .await?
+            .into_iter()
+            .map(|repository| RepositoryConfig::new_named(repository.name, repository.git_url))
+            .collect();
+        let mut repository_lookup = self.repository_lookup.write().unwrap();
         debug!("Reloading repositoriy metadata...");
-        *repositories = load_meta();
+        *repository_lookup = load_meta();
```

Comment:
```suggestion
        *repository_lookup = load_meta();
```

https://github.com/TabbyML/tabby/blob/0f22fddad33134059bf40306308eee1e3bc56d2a/ee/tabby-webserver/src/repositories/resolve.rs#L162

Load meta should read from repositories list from `RepositoriyService` as well